### PR TITLE
Fix AppVeyor build, add tests results uploading and build version patching.

### PR DESCRIPTION
**Pull request by gdziadkiewicz**
I didn't manage to make the documentation generation run on AppVeyor, but even building and running tests on a Windows box seems beneficial.

@gusty , if you decide to merge that can you take care of turning on the AppVeyor build and adding the badge to the readme?

Link to passing build on my AppVeyor account:
https://ci.appveyor.com/project/gdziadkiewicz/fsharpplus/builds/19799286

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -2,9 +2,8 @@ image: Visual Studio 2017
 init:
   - git config --global core.autocrlf input
 build_script:
-  - cmd: build.cmd All
```
Comment (wallymathieu) [inline appveyor.yml]: Doesn't the other targets work?

**Comment by wallymathieu** [comment]
Perhaps @forki or someone else with fsharproject admin access can enable appveyor build?

**Comment by wallymathieu** [comment]
I'm going to close and reopen to trigger appveyor build

**Comment by gusty** [comment]
It would be really interesting to have the docs building in a CI.
This way when a change is required in the docs we can detect a failure, I would like to have at least a step that tried to compile the scripts and fails the whole builds if there's a compilation error.

The error [I see here](https://ci.appveyor.com/project/wallymathieu/fsharpplus/builds/19800452) makes me think it's trying to compile those scripts with FSharp.Core 3, where `Result<_,_>` was not available.

**Pull request merged**